app/src/androidTest/java/com/github/ybecker/epforuml/database/FirebaseDatabaseAdapterTest.kt

```diff
@@ -123,6 +134,36 @@ class FirebaseDatabaseAdapterTest {
         db.addChat(romain.userId, theo.userId ,"Hi Theo this is Romain!")
     }
 
+    @Test
+    fun AAAaddMessageRefresh() {
+        Thread.sleep(10000) // wait for all futures to complete
```

Comment:
Even if you want to wait every future, I think that the Thread.sleep is too long !